constellation-app/constellation
# Conversation Unresponsive Bug Fix

### Prerequisites


    for the "CI / Unit Tests") to complete first. Failed Unit tests can be 
    debugged by adding the label "verbose logging" to the GitHub PR.

### Description of the Change
Added a Platform.runLater() to when the content values are processed which stops the unexpected exception from throwing.
Added a page system for viewing the conversation bubbles as viewing less at once lowers memory usage and speeds up the view.


### Alternate Designs



### Why Should This Be In Core?
Bugfix in Core and enhancement to Core functionality.


### Benefits
Makes it easier to navigate lots of Conversation bubbles.


### Possible Drawbacks



### Verification Process
For testing the unexpected exception bug:
1. Create a new graph. Build a sphere graph with 50 nodes and 50000 transactions
2. Select all transactions. Add the attributes Content and Content.Translated in the Attribute Editor. Give both of these attributes values (make sure all transactions have values)
3. Deselect everything on the graph. Open the Conversation View
4. Select a node on the graph. Unexpected exception should no longer be thrown


For testing the pages:
1. Open or create a graph with lots of Conversation content
2. Select a node on the graph
3. Update the Content Per Page dropdown at the top of the view to change how many bubbles are visible per page 
4. Select the numbers at the bottom of the view to change pages. If it just shows 1 page, then all bubbles connected to that node or transaction fit on one page.


### Applicable Issues
#1988
